frontend/src/utilities/useWatchIntegrationComponents.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import { useAppSelector } from '~/redux/hooks';
+import { OdhApplication } from '~/types';
+import { getIntegrationAppEnablementStatus } from '~/services/integrationAppService';
+import { allSettledPromises } from '~/utilities/allSettledPromises';
+import { POLL_INTERVAL } from './const';
+import { isIntegrationApp } from './utils';
+
+type OdhIntegrationApplication = OdhApplication & { spec: { internalRoute: string } };
```

Comment:
Can you move this into the `types.ts` next to `OdhApplication`.

Something like this:
```
/**
 * An OdhApplication that uses integration api to determine status.
 * @see isIntegrationApp
 */
type OdhIntegrationApplication = OdhApplication & {
  spec: {
    internalRoute: string; // starts with `/api/`
  }
};
```

Reply:
Moved